pkg/reconciler/taskrun/validate_resources_test.go

```diff
@@ -155,6 +155,13 @@ func TestValidateResolvedTaskResources_ValidParams(t *testing.T) {
 						"key1": {},
 						"key2": {},
 					},
+					Default: &v1beta1.ParamValue{
```

Comment:
again could you keep the existing test case and add a new one?

Reply:
This newly added test `myObjWithDefault` might be also the one that you mentioned "we might still want to validate for taskruns with inline task specs that all the keys have values defined somewhere." It is simulating the taskspec has 3 keys and default provides a value for `key1` and `key3`, taskrun provides a value for `key2` and `key3`.

Reply:
I was actually thinking we could validate in the admission webhook in addition to the validation that exists in the reconciler for taskruns with inline task specs. This would mean the admission webhook can reject the taskrun instead of having the reconciler fail the taskrun. but what you have here is OK

Reply:
Aha I see what you mean. Thanks for clarification.
Yeah, the validation against params across task and taskrun is done on the reconciler side specifically `pkg/reconciler/taskrun/validate_resources.go` file i.e. checking if a param `foo` of string type declared in task spec is provided with a value of an array type, checking missing params and missing keys for object param across default and taskrun value etc.